qrand/platforms/cirq/circuit.py

```diff
@@ -19,85 +19,117 @@
 
     def x(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.X(self.qubits[target_qubit]))
+
 
     def y(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Y(self.qubits[target_qubit]))
+
 
     def z(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self.moment_list.append(cirq.ops.Z(self.qubits[target_qubit]))
 
     ############################# TWO QUBIT GATES #############################
-    def cs(self, control_qubit: int, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
-    def cx(self, control_qubit: int, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
-
     def cz(self, control_qubit: int, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit, True)
+        validate_natural_number(control_qubit, True)
+        self._validate_qubit_index(target_qubit)
+        self._validate_qubit_index(control_qubit)
+        self.moment_list.append(cirq.ops.CZ(self.qubits[control_qubit], self.qubits[target_qubit]))
 
     def swap(self, target_qubit_1: int, target_qubit_2: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit_1, True)
+        self._validate_qubit_index(target_qubit_1)
+        self._validate_qubit_index(target_qubit_2)
+        self.moment_list.append(cirq.ops.SWAP(self.qubits[target_qubit_1], self.qubits[target_qubit_2]))
 
     ############################ THREE QUBIT GATES ############################
-    def ccx(
-        self, control_qubit_1: int, control_qubit_2: int, target_qubit: int
-    ) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+
 
     def cswap(
-        self, control_qubit_1: int, control_qubit_2: int, target_qubit: int
+        self, control_qubit: int, target_qubit_1: int, target_qubit_2: int
     ) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        validate_natural_number(target_qubit_1, True)
+        validate_natural_number(target_qubit_2, True)
+        validate_natural_number(control_qubit, True)
+        self._validate_qubit_index(control_qubit)
+        self._validate_qubit_index(target_qubit_1)
+        self._validate_qubit_index(target_qubit_2)
+        self.moment_list.append(cirq.ops.CSWAP(self.qubits[control_qubit], self.qubits[target_qubit_1], self.qubits[target_qubit_2]))
+
+    def createcircuit(self):
+        return CirqQuantumCircuit(self.moment_list)
```

Comment:
It is better to inherit from `CirqQuantumCircuit` and apply the gates to itself: just as it is done in `QiskitCircuit`. This other technique should be avoided. Therefore you do not need the `moment_list` attribute.

Reply:
Will do it.